src/NodaTime/Calendars/Wondrous/WondrousCalendarHelper.cs

```diff
@@ -0,0 +1,2204 @@
+        ///     - treat Ayyam-i-Ha as month 20. This will cause problems if months are sorted naively.
+        ///     - treat Ayyam-i-Ha as extra days in month 18. This would require special handling to sometimes display
+        ///       days in "month 18" as being Ayyam-i-Ha, but we have no direct control over display.
+        ///       
+        /// This version will treat Ayyam-i-Ha days as extra days in month 18.
+        /// 
+        /// Users interacting with the system should use 0 as Ayyam-i-Ha.
+        /// </remarks>
+        internal const int AyyamiHaMonth0 = 0;
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="wondrousYear">Year in the Wondrous calendar</param>
+        /// <param name="wondrousMonth">Month (use 0 for Ayyam-i-Ha)</param>
+        /// <param name="wondrousDay">Day in month</param>
+        /// <returns></returns>
+        public static LocalDate CreateDate(int wondrousYear, int wondrousMonth, int wondrousDay)
+        {
+            if (wondrousMonth == AyyamiHaMonth0)
+            {
+                var maxDay = GetYearInfo(wondrousYear).DaysInAyyamiHa;
+
+                Preconditions.CheckArgumentRange(nameof(wondrousDay), wondrousDay, 1, maxDay);
+
+                // move Ayyam-i-Ha days to fall after the last day of month 18.
+                wondrousMonth = Month18;
+                wondrousDay += DaysInMonth;
+            }
+            return new LocalDate(wondrousYear, wondrousMonth, wondrousDay, CalendarSystem.Wondrous);
+        }
+
+        /// <summary>
+        /// Create a date in the Wondrous calendar.
+        /// </summary>
+        /// <param name="wondrousYear"></param>
+        /// <param name="wondrousMonth"></param>
+        /// <param name="wondrousDay"></param>
+        /// <param name="hour"></param>
+        /// <param name="minute"></param>
+        /// <param name="second"></param>
+        /// <param name="millisecond"></param>
+        /// <returns></returns>
+        public static LocalDateTime CreateDateTime(int wondrousYear, int wondrousMonth, int wondrousDay, int hour, int minute = 0, int second = 0, int millisecond = 0)
+        {
+            if (wondrousMonth == AyyamiHaMonth0)
+            {
+                // move Ayyam-i-Ha days to fall after the last day of month 18.
+                wondrousMonth = Month18;
+                wondrousDay += DaysInMonth;
+            }
+            return new LocalDateTime(wondrousYear, wondrousMonth, wondrousDay, hour, minute, second, millisecond, CalendarSystem.Wondrous);
+        }
+
+        /// <summary>
+        /// Return the month of this date. If in Ayyam-i-Ha, returns 0.
+        /// </summary>
+        /// <remarks>Deals with Ayyam-i-Ha.</remarks>
+        /// <param name="input"></param>
+        /// <returns></returns>
+        public static int WondrousMonthNum(this LocalDate input)
```

Comment:
I'd suggest calling this `GetWondrousMonth` - the `Num` part is definitely non-idiomatic.
(Ditto following methods.)

Probably worth throwing around multiple ideas before you implement the change though.

Reply:
Adding "Get" sounds good. And I agree, "Num" is not needed as there is no other meaning for Month in this area.

Ideally, the "this" type would be WondrousCalendar, but that's not how the system works.